### PR TITLE
Removes _index from cocos params, improves & fixes docs, raises when cannot det. single cocos or no qpsi present

### PR DESCRIPTION
**Pull request by oliverfunk**

### CODE REVIEW

File: eqdsk/file.py [hunk 1]
```diff
@@ -212,64 +234,93 @@ def cocos(self) -> COCOS:
...
         volt_seconds_per_radian: bool | None = None,
+        qpsi_sign: Sign | None = None,
     ):
-        """Identify the COCOS of this eqdsk and set the COCOS attribute.
+        """Identifies the COCOS of this eqdsk.
+
+        Note
+        ----
+            This sets the internal _cocos attribute and does not return
+            anything.
```
Comment (je-cook) [inline eqdsk/file.py:248]:
```suggestion
        This sets the internal _cocos attribute and does not return
        anything.
```

File: tests/test_file.py [hunk 2]
```diff
@@ -84,15 +79,13 @@ def test_no_cocos_access_raise_ValueError(self):
         ("file", "ftype", "ind"),
         [
             ("jetto.eqdsk_out", "eqdsk", 11),
-            ("DN-DEMO_eqref.json", "json", 17),
-            ("eqref_OOB.json", "json", 17),
-            ("DN-DEMO_eqref_withCoilNames.json", "json", 17),
+            ("DN-DEMO_eqref.json", "json", 7),
```
Comment (je-cook) [inline tests/test_file.py:82]: this is 3 I think (there is no qpsi), or 7 if you want

File: tests/test_file.py [hunk 3]
```diff
@@ -84,15 +79,13 @@ def test_no_cocos_access_raise_ValueError(self):
...
+            ("eqref_OOB.json", "json", 7),
+            ("DN-DEMO_eqref_withCoilNames.json", "json", 7),
```
Comment (je-cook) [inline tests/test_file.py:84]: Same here cocos 3

File: tests/test_file.py [hunk 4]
```diff
@@ -119,7 +112,7 @@ def test_read_wrong_cocos_raises_ValueError(self):
     @pytest.mark.parametrize(
         ("setup_keys", "end_keys"),
         [
-            ({"from_cocos_index": 17, "to_cocos_index": 11}, {"from_cocos_index": 11}),
+            ({"from_cocos": 17, "to_cocos": 11}, {"from_cocos": 11}),
```
Comment (je-cook) [inline tests/test_file.py:115]: I dont get how this one can pass as its probably 3

File: tests/test_file.py [hunk 5]
```diff
@@ -158,7 +151,7 @@ def test_write_with_wrong_length_qsi_raises_ValueError(self, tmp_path):
         file = self.data_dir / "DN-DEMO_eqref.json"
         # Create EQDSK file interface and read data to a dict
 
-        eqdsk = EQDSKInterface.from_file(file, from_cocos_index=17, to_cocos_index=11)
+        eqdsk = EQDSKInterface.from_file(file, from_cocos=17, to_cocos=11)
```
Comment (je-cook) [inline tests/test_file.py:154]: again should be 3

**Review by je-cook — CHANGES_REQUESTED** [review]
We should probably detect the eqdsks correctly inc the ones in private data. I think they're either 3, 7 or 11 no 17s

**Review by je-cook — APPROVED** [review]
I like how if you provide 0 data it treats it as unset. We should make sure that is in the docs when we build them!

**Review by OceanNuclear — APPROVED** [review]
Looks good

**Pull request merged**